awspricing/__init__.py

```diff
@@ -28,69 +24,48 @@ def _fetch_offers():
     if offers is not None:
         return offers
 
-    resp = session.get(OFFER_BASE_URL + OFFER_INDEX_ENDPOINT)
-    resp.raise_for_status()
-    offers = resp.json()['offers']
+    for product in DEFAULT_AWS_PRODUCTS:
+        _fetch_offer(product)
 
     maybe_write_to_cache(cache_key, offers)
     return offers
 
 
-def _fetch_available_versions(offers, offer_name):
-    cache_key = 'offers_version_{}'.format(offer_name)
-    offers_version = maybe_read_from_cache(cache_key)
-    if offers_version is not None:
-        return offers_version
+def _get_services():
+    global service_list
+    if not service_list:
+        service_list = all_services_names()
+    return service_list
 
-    offers_version_endpoint = offers[offer_name]['versionIndexUrl']
-    resp = session.get(OFFER_BASE_URL + offers_version_endpoint)
-    resp.raise_for_status()
-    offers_version = resp.json()['versions']
 
-    maybe_write_to_cache(cache_key, offers_version)
-    return offers_version
-
-
-def _get_offers():
-    global _OFFERS
-    if not _OFFERS:
-        _OFFERS = _fetch_offers()
-    return _OFFERS
-
-
-def _fetch_offer(offer_name, version='current'):
-    offers = _get_offers()
-    if offer_name not in offers:
-        raise ValueError('Unknown offer name: {}'.format(offer_name))
+def _fetch_offer(offer_name, version=datetime.datetime.now().strftime(TIME_FORMAT)):
+    services = _get_services()
+    if offer_name not in services:
+        raise ValueError('Unknown offer name, no corresponding AWS Service: {}'.format(offer_name))
 
     cache_key = 'offer_{}_{}'.format(offer_name, version)
     offer = maybe_read_from_cache(cache_key)
     if offer is not None:
         return offer
 
-    if version == 'current':
-        offer_endpoint = offers[offer_name]['currentVersionUrl']
-    else:
-        other_versions = _fetch_available_versions(offers, offer_name)
-        if version in other_versions:
-            offer_endpoint = other_versions[version]['offerVersionUrl']
-        else:
-            raise ValueError('Invalid version specified {}. Must be one of: [{}]'.format(
-                version, ', '.join(sorted(other_versions.keys(), reverse=True))))
-
-    resp = session.get(OFFER_BASE_URL + offer_endpoint)
-    resp.raise_for_status()
-    offer = resp.json()
+    paginator = client.get_paginator('get_products')
+    resp_pages = paginator.paginate(ServiceCode=offer_name, FormatVersion='aws_v1')
+    offer = []
+    for page in resp_pages:
+        for product in page['PriceList']:
+            offer.append(ast.literal_eval(product))
```

Comment:
why use `ast` here instead of `json.loads`?

Reply:
Did some reading and `json.loads` seems safer - changed to `json.loads`